application/src/main/java/org/togetherjava/tjbot/features/jshell/JShellCommand.java

```diff
@@ -0,0 +1,298 @@
+                            .addOption(OptionType.STRING, JSHELL_CODE_PARAMETER,
+                                    "Code to evaluate. If not supplied, open an inout box.")
+                            .addOption(OptionType.BOOLEAN, JSHELL_STARTUP_SCRIPT_PARAMETER,
+                                    "If the startup script should be loaded, true by default."),
+                new SubcommandData(JSHELL_SNIPPETS_SUBCOMMAND,
+                        "Get the evaluated snippets of the user who sent the command, or the user specified user if any.")
+                            .addOption(OptionType.USER, JSHELL_USER_PARAMETER,
+                                    "User to get the snippets from. If null, get the snippets of the user who sent the command.")
+                            .addOption(OptionType.BOOLEAN, JSHELL_INCLUDE_STARTUP_SCRIPT_PARAMETER,
+                                    "if the startup script should be included, false by default."),
+                new SubcommandData(JSHELL_CLOSE_SUBCOMMAND, "Close your session."),
+                new SubcommandData(JSHELL_STARTUP_SCRIPT_SUBCOMMAND,
+                        "Display the startup script."));
+    }
+
+    @Override
+    public void onSlashCommand(SlashCommandInteractionEvent event) {
+        switch (Objects.requireNonNull(event.getSubcommandName())) {
+            case JSHELL_VERSION_SUBCOMMAND -> handleVersionCommand(event);
+            case JSHELL_EVAL_SUBCOMMAND -> handleEvalCommand(event);
+            case JSHELL_SNIPPETS_SUBCOMMAND -> handleSnippetsCommand(event);
+            case JSHELL_CLOSE_SUBCOMMAND -> handleCloseCommand(event);
+            case JSHELL_STARTUP_SCRIPT_SUBCOMMAND -> handleStartupScriptCommand(event);
+            default -> throw new AssertionError(
+                    "Unexpected Subcommand: " + event.getSubcommandName());
+        }
+    }
+
+    @Override
+    public void onModalSubmitted(ModalInteractionEvent event, List<String> args) {
+        ModalMapping mapping =
+                event.getValue(JSHELL_TEXT_INPUT_ID + "|" + JSHELL_STARTUP_SCRIPT_PARAMETER);
+        boolean startupScript = mapping != null;
+        if (mapping == null) {
+            mapping = event.getValue(JSHELL_TEXT_INPUT_ID);
+        }
+        if (mapping != null) {
+            handleEval(event, event.getUser(), true, mapping.getAsString(), startupScript);
+        }
+    }
+
+    private void handleVersionCommand(SlashCommandInteractionEvent event) {
+        String code = """
+                System.out.println("```");
+                System.out.println("Version: " + Runtime.version());
+                System.out.println("Vendor:  " + System.getProperty("java.vendor"));
+                System.out.println("OS:      " + System.getProperty("os.name"));
+                System.out.println("Arch:    " + System.getProperty("os.arch"));
+                 System.out.println("```");""";
```

Comment:
the last line has an extra space, looks odd

Reply:
Fixed.